lib/Sema/TypeCheckDecl.cpp

```diff
@@ -2223,12 +2224,20 @@ InterfaceTypeRequest::evaluate(Evaluator &eval, ValueDecl *D) const {
 
   case DeclKind::Var: {
     auto *VD = cast<VarDecl>(D);
-    auto *namingPattern = VD->getNamingPattern();
-    if (!namingPattern) {
+    Type interfaceType;
+    if (auto *parentE = VD->getParentExpr()) {
+      // Type check the ASTNode that contains the parent 'TapExpr' if it
+      // hasn't.
+      if (!parentE->getType())
+        swift::typeCheckASTNodeAtLoc(VD->getDeclContext(), parentE->getLoc());
```

Comment:
This is subject to the same problems we discussed before. Performing a full type check is not appropriate in the interface type request, and *will* cause circularity problems to be surfaced as in valid entries in Code Completion. Consider a variable binding whose initializer makes reference to the variable itself. Computing the interface type of the declaration doesn't necessarily need to typecheck the initializer binding if there is a user-provided contextual type. Now, it will.

Reply:
Is this can be handled at the top-level as closures can, that would be great.

Reply:
OK, I'm going to give up on-demand type checking for `TapExpr` for now.